flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/reader/deserializer/PulsarSchemaWrapper.java

```diff
@@ -41,22 +43,45 @@
     /** The serializable pulsar schema, it wrap the schema with type class. */
     private final PulsarSchema<T> pulsarSchema;
 
+    private boolean isSchemaEvolutionEnabled;
+
     public PulsarSchemaWrapper(PulsarSchema<T> pulsarSchema) {
         this.pulsarSchema = pulsarSchema;
     }
 
     @Override
-    public void deserialize(Message<byte[]> message, Collector<T> out) throws Exception {
-        Schema<T> schema = this.pulsarSchema.getPulsarSchema();
-        byte[] bytes = message.getData();
-        T instance = schema.decode(bytes);
+    public void open(
+            DeserializationSchema.InitializationContext context, SourceConfiguration configuration)
+            throws Exception {
+        this.isSchemaEvolutionEnabled = configuration.isEnableSchemaEvolution();
+    }
 
-        out.collect(instance);
+    @Override
+    public void deserialize(Message<?> message, Collector<T> out) throws Exception {
```

Comment:
I don't like this kind of deserialization.

Reply:
Would you mind suggest an alternative ? I remember we discussed earlier and this is the discussed design, maybe I misundertood ?